test/transcoding/cl-types.ll

```diff
@@ -102,21 +103,24 @@ define spir_kernel void @foo(
   %opencl.image1d_buffer_t addrspace(1)* nocapture %g1,
   %opencl.image1d_t addrspace(1)* nocapture %c2,
   %opencl.image2d_t addrspace(1)* nocapture %d3,
-  i32 %s) #0 !kernel_arg_addr_space !1 !kernel_arg_access_qual !2 !kernel_arg_type !3 !kernel_arg_base_type !4 !kernel_arg_type_qual !5 {
+  %opencl.sampler_t* %s) #0 !kernel_arg_addr_space !1 !kernel_arg_access_qual !2 !kernel_arg_type !3 !kernel_arg_base_type !4 !kernel_arg_type_qual !5 {
 entry:
 ; CHECK-SPIRV: 5 SampledImage [[SAMPIMG]] [[SAMPIMG_VAR1:[0-9]+]] [[IMG_ARG]] [[SAMP_ARG]]
 ; CHECK-SPIRV: 7 ImageSampleExplicitLod {{[0-9]+}} {{[0-9]+}} [[SAMPIMG_VAR1]]
-; CHECK-LLVM:   call spir_func <4 x float> @_Z11read_imagef11ocl_image2d11ocl_samplerDv4_if(%opencl.image2d_t addrspace(1)* %d1, i32 %s, <4 x i32> zeroinitializer, float 1.000000e+00)
-  %.tmp = call spir_func <4 x float> @_Z11read_imagef11ocl_image2d11ocl_samplerDv4_if(%opencl.image2d_t addrspace(1)* %d1, i32 %s, <4 x i32> zeroinitializer, float 1.000000e+00)
+; CHECK-LLVM:   call spir_func <4 x float> @_Z11read_imagef11ocl_image2d11ocl_samplerDv4_if(%opencl.image2d_t addrspace(1)* %d1, %opencl.sampler_t* %s, <4 x i32> zeroinitializer, float 1.000000e+00)
+  %.tmp = call spir_func <4 x float> @_Z11read_imagef11ocl_image2d11ocl_samplerDv4_if(%opencl.image2d_t addrspace(1)* %d1, %opencl.sampler_t* %s, <4 x i32> zeroinitializer, float 1.000000e+00)
 
 ; CHECK-SPIRV: 5 SampledImage [[SAMPIMG]] [[SAMPIMG_VAR2:[0-9]+]] [[IMG_ARG]] [[SAMP_CONST]]
 ; CHECK-SPIRV: 7 ImageSampleExplicitLod {{[0-9]+}} {{[0-9]+}} [[SAMPIMG_VAR2]]
-; CHECK-LLVM:   call spir_func <4 x float> @_Z11read_imagef11ocl_image2d11ocl_samplerDv4_if(%opencl.image2d_t addrspace(1)* %d1, i32 32, <4 x i32> zeroinitializer, float 1.000000e+00)
-  %.tmp2 = call spir_func <4 x float> @_Z11read_imagef11ocl_image2d11ocl_samplerDv4_if(%opencl.image2d_t addrspace(1)* %d1, i32 32, <4 x i32> zeroinitializer, float 1.000000e+00)
+; CHECK-LLVM:   call spir_func <4 x float> @_Z11read_imagef11ocl_image2d11ocl_samplerDv4_if(%opencl.image2d_t addrspace(1)* %d1, %opencl.sampler_t* %0, <4 x i32> zeroinitializer, float 1.000000e+00)
+  %0 = call %opencl.sampler_t* @__translate_sampler_initializer(i32 32)
```

Comment:
Please add a check to make sure this call is preserved after round trip translation.

Reply:
Done.